src/scenes/Calendar.tsx

```diff
@@ -1,29 +1,65 @@
-import React from 'react';
-import {Calendar as CalendarComponent} from 'antd';
-import { Moment } from 'moment';
+import React, {useState} from 'react';
+import {Calendar as CalendarComponent, Card, Modal} from 'antd';
+import moment, { Moment } from 'moment';
 import {useAppSelector} from "../redux/store";
 import {Meeting, meetingsSelectors} from "../redux/meetingsSlice";
+import {Assignment, assignmentsSelectors} from "../redux/assignmentSlice";
+import {withRouter, Link} from "react-router-dom";
 
-const Calendar = () => {
+const Calendar = (props: any) => {
     const meetings = useAppSelector(meetingsSelectors.selectAll);
+    const assignments = useAppSelector(assignmentsSelectors.selectAll);
+
+    const [isModalVisible, setIsModalVisible] = useState(false);
+    const [selectedDate, setSelectedDate] = useState('');
 
     const dateCellRender = (dateCellMoment: Moment) => {
         const meetingsOnDate: Meeting[] = meetings.filter(meeting => {
             return dateCellMoment.format('YYYY-MM-DD') === meeting.date
         });
 
-        if (!meetingsOnDate) {
+        const assignmentsOnDate: Assignment[] = assignments.filter(assignment => {
+            return dateCellMoment.format('YYYY-MM-DD') === moment(assignment.due_date).format('YYYY-MM-DD')
+        });
+
+        if (!meetingsOnDate && !assignmentsOnDate) {
             return null;
         }
 
         return (
             <ul>
                 {meetingsOnDate.map(meeting => <li>{meeting.name}</li>)}
+                {assignmentsOnDate.map(assignment => <li>{assignment.name}</li>)}
             </ul>
         )
     };
 
-    return <CalendarComponent dateCellRender={dateCellRender} />;
+    const onSelect = (selectedDate: Moment) => {
+        setIsModalVisible(true);
+        setSelectedDate(selectedDate.format('YYYY-MM-DD'))
+    }
+
+    return (
+        <>
+            <Modal title={moment(selectedDate).format('dddd[,] MMMM Do')} visible={isModalVisible} onOk={()=>setIsModalVisible(false)} onCancel={() => setIsModalVisible(false)}>
+                    <h3>Assignments</h3>
+                    {assignments.filter(assignment => {
+                        return selectedDate === moment(assignment.due_date).format('YYYY-MM-DD');
+                    })
+                        .map(assignment => {
+                            return <Link to={`/assignments/${assignment.id}`}><Card hoverable><p><b>{assignment.name}</b></p><p>Due at {moment(assignment.due_date).format('hh:mma')}</p></Card></Link>
```

Comment:
Could you format this so it's a bit more readable? Maybe make it multiline, or make a function?

Same thing on lines 44 and 57.